src/Test/StateMachine/Parallel.hs

```diff
@@ -648,25 +648,26 @@ executeParallelCommands :: (Show (cmd Concrete), Show (resp Concrete))
                         -> ParallelCommands cmd resp
                         -> UIO.TChan (Pid, HistoryEvent cmd resp)
                         -> Bool
-                        -> m (History cmd resp, model Concrete, Reason, Reason)
-executeParallelCommands sm@StateMachine{ initModel } (ParallelCommands prefix suffixes) hchan stopOnError = do
+                        -> m (History cmd resp, model Concrete, Reason, Reason, Property)
+executeParallelCommands sm@StateMachine{ initModel, finalCheck } (ParallelCommands prefix suffixes) hchan stopOnError = do
 
-  (reason0, (env0, _smodel, _counter, cmodel)) <- runStateT
+  ((reason0, prop0), (env0, _smodel, _counter, cmodel)) <- runStateT
       (executeCommands sm hchan (Pid 0) CheckEverything prefix)
       (emptyEnvironment, initModel, newCounter, initModel)
   if reason0 /= Ok
   then do
     hist <- getChanContents hchan
-    return (History hist, cmodel, reason0, reason0)
+    return (History hist, cmodel, reason0, reason0, prop0)
   else do
     (reason1, reason2, _) <- go (Ok, Ok, env0) suffixes
     hist <- getChanContents hchan
-    return (History hist, cmodel, reason1, reason2)
+    prop <- fromMaybe (property True) <$> finalCheck
+    return (History hist, cmodel, reason1, reason2, prop)
```

Comment:
I have to re-read this part, which is where the important stuff is changing. I am unsure if it works as expected. I think we don't want to use `prop0` and we instead want to always check `finalCheck`

Reply:
(in fact, `prop0` necessarily has to be `finalCheck` again)